src/ameba/rule/typing/proc_literal_return_type_restriction.cr

```diff
@@ -0,0 +1,43 @@
+module Ameba::Rule::Typing
+  # A rule that enforces that Proc literals have a return type.
+  #
+  # For example, these are considered valid:
+  #
+  # ```
+  # my_proc = ->(arg : String) : String { a + "proc" }
+  # ```
+  #
+  # ```
+  # task -> : Task { Task.new("execute this command") }
+  # ```
+  #
+  # And these are invalid:
+  #
+  # ```
+  # my_proc = ->(arg : String) { a + "proc" }
+  # ```
+  #
+  # ```
+  # task -> { Task.new("execute this command") }
+  # ```
+  #
+  # YAML configuration example:
+  #
+  # ```
+  # Typing/ProcReturnTypeRestriction:
+  #   Enabled: true
```

Comment:
taking into account that all the other suggested rules in Typing scope are disabled, should this one be enabled?

Reply:
This case IIRC (unlike the others) can lead to footguns when passing procs to/from C, and you already need to define the param types for procs. I can go either way though

Reply:
I'd keep it disabled for now. Passing procs to/from C is a) not that common, b) you don't *need* the annotation, so IMO it's more up to the project to enable these.